### PR TITLE
Issue #203: use okhttp3/retrofit2/moshi

### PR DESCRIPTION
**Pull request by bentatham**
Here is an update from old okhttp/retrofit to okhttp3/retrofit2/moshi.  I chose moshi (over, say, gson) since it is my understanding that since is also uses okio, it is the most memory efficient of the converters.

Logging moved from retrofit to okhttp, using an interceptor.

Error handling has changed too, and is now directly handled by exceptions instead.

I can't run docker on my pc right now due to the need for virtualbox for some other dev I'm doing, but installed influxed locally and it all passes.  The one change might be the user/pass combo - not sure if the docker container has different default passwords for some reason.

### CODE REVIEW

**Comment by majst01** [comment]
Excellent !
I will take some time to review, but so far looks good.

File: src/main/java/org/influxdb/impl/InfluxDBImpl.java [hunk 1]
```diff
@@ -175,59 +198,73 @@ public void write(final String database, final String retentionPolicy, final Con
 	 * {@inheritDoc}
 	 */
 	@Override
-	public QueryResult query(final Query query) {
-		QueryResult response = this.influxDBService
-				.query(this.username, this.password, query.getDatabase(), query.getCommand());
-		return response;
+	public QueryResult query(final Query query)	{
+		return execute(query.requiresPost()
+				? this.influxDBService.postQuery(this.username, this.password, query.getDatabase(), query.getCommand())
```
Comment (majst01) [inline src/main/java/org/influxdb/impl/InfluxDBImpl.java]: Please do not use ternary form, convert to more readable if then else form.

**Comment by majst01** [comment]
We should also add a notice in the Readme that beginning with 2.4 retrofit2/okhttp3 is used and the api usage differs

**Comment by bentatham** [comment]
Given the API diff, should it be 3.0?  I defer to you on that of course.

**Comment by majst01** [comment]
Hi @bentatham Thanks for your work, i think this is ready to be merged now but i would get feedback from my co-maintainer as well.

@andrewdodd  Can you please have a look at the MR and comment.

**Comment by majst01** [comment]
As Andrew did not respond i am merging this now.

**Pull request merged**